### PR TITLE
Add macs conversion from msp

### PR DESCRIPTION
**Pull request by winni2k**
Move the macs and newick CLI commands over from msprime. See https://github.com/tskit-dev/msprime/pull/1124. This closes https://github.com/tskit-dev/msprime/issues/1120.

### CODE REVIEW

**Comment by jeromekelleher** [comment]
Thanks @winni2k. I'm not sure there's much point in offering these as parts of the ``tskit`` cli though (it's surely a pretty niche requirement). I wonder if it would be better to add a method ``as_macs`` (or ``to_macs``, maybe?) to the TreeSequence class. This would go well with the proposed ``as_ms`` method (#1120). It also makes me wonder if we should call the current ``nexus`` method ``as_nexus`` - this would give a nice convention for converting to other formats. What do you think @winni2k?

Certainly we don't need the CLI ``newick`` command anyway, I think, so we can just drop that.

**Comment by winni2k** [comment]
Ah, dang, I missed the link to #727. 

If we have a nexus converter (where?) then the newick conversion is certainly obsolete. I also see we have newick conversion in ts_convert (https://github.com/tskit-dev/tsconvert/blob/master/tsconvert/newick.py#L136). There's also a `to_ms` method in ts_convert? https://github.com/tskit-dev/tsconvert/blob/master/tsconvert/newick.py#L105 

We are using `to_newick` in ts_convert, so maybe go with `to_*`? 

Removing the newick conversion from tskit now...

**Comment by jeromekelleher** [comment]
Great, thanks @winni2k. I didn't explain myself very well the last time, but I was suggesting that we get rid of the ``to_macs`` CLI command and instead add a ``to_macs`` (or ``as_macs`` - I'm still on the fence on this one!) *method* to the TreeSequence class. I don't think we need the CLI end point, but a method could be useful to people. It's basically the same code, it's just the tests will live in a different module (I guess the ``test_highlevel.py`` file).

I wouldn't worry too much about what things are called in tsconvert - that's never really gotten off the ground and was never brought up to a release.

**Comment by winni2k** [comment]
Ah, conversion commands on the CLI would clutter the tskit interface? 

I suppose the correct thing to do then would be to move the CLI command to ts_convert, but have the method on the TreeSequence class. I'll add the method. 

Naming: How about following the example of pandas dataframe conversion on this? https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.to_csv.html

**Comment by winni2k** [comment]
Do you have a link for the macs format so that I can write something sensible in the method documentation?

**Comment by winni2k** [comment]
Also, what about `write_macs`? We have the `write_vcf` method...

**Comment by jeromekelleher** [comment]
> Naming: How about following the example of pandas dataframe conversion on this?

Yes! Perfect - I like having someone else to follow. Let's call them ``to_X``. We can make things more consistent later, let's just get the ``to_macs`` method in for now.

I think a link to the [source repo](https://github.com/gchen98/macs) for macs is good enough in terms of docs.

File: python/tskit/trees.py [hunk 1]
```diff
@@ -4295,6 +4295,34 @@ def nexus(self, precision=14):
         s += "END;\n"
         return s
 
+    def to_macs(self, path_or_buf=None):
```
Comment (jeromekelleher) [inline python/tskit/trees.py:4298]: We're not testing the ``path_or_buf`` argument here, so I'd say let's not bother with it. Just return a string - we can add this type of argument later, if we need it (and should try to be consistent across the formats).

**Comment by jeromekelleher** [comment]
Great, thanks @winni2k - can you squash the commits please? Good to merge then.

**Comment by jeromekelleher** [comment]
Excellent, thanks @winni2k! Sorry, I forgot one thing - can you add this to the python CHANGELOG in the "New features" section please (I should add a tickbox for this in a PR template, I keep forgetting about it!)

**Comment by winni2k** [comment]
Of course! How about this?

**Review by jeromekelleher — APPROVED** [review]

**Comment by jeromekelleher** [comment]
Perfect! Thanks @winni2k !

**Pull request merged**